lib/data/multipleBackendGateway.js

```diff
@@ -201,7 +201,22 @@ const multipleBackendGateway = {
                     return cb(err);
                 }
                 return client.uploadPart(request, streamingV4Params, stream,
-                    size, key, uploadId, partNumber, bucketName, log, cb);
+                size, key, uploadId, partNumber, bucketName, log,
+                (err, partInfo) => {
+                    if (err) {
+                        // if error putting part, counter should be decremented
+                        return locationStorageCheck(location, -size, log,
+                        error => {
+                            if (error) {
+                                log.error('Error decrementing location ' +
```

Comment:
Let's stick with the general logging practice for errors: `log.error('some message', { error: err.message })`